arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/RaiseAccumulate.kt

```diff
@@ -506,25 +506,25 @@ public inline fun <Error, A, B, C, D, E, F, G, H, I, J> Raise<NonEmptyList<Error
 public inline fun <Error, A> Raise<Error>.forEachAccumulating(
   iterable: Iterable<A>,
   combine: (Error, Error) -> Error,
-  @BuilderInference block: RaiseAccumulate<Error>.(A) -> Unit
+  @BuilderInference block: RaiseAccumulate<Error>.(hasError: Boolean, A) -> Unit
```

Comment:
Can we do this in a way it doesn't leak in the public implementation 😢 Leaking such low-level optimisations into the public API is a big red flag for me. I personally prefer the memory hit :/

Reply:
I figured having that accessible could be useful for callers. However, yes there's a way to do it by just wrapping every call in a `try {} catch(RaiseCancellationException)` and rethrowing it. Would that be preferable to you?

Reply:
As a general suggestion, I would suggest having extreme caution with `inline` methods. It's not that we are just leaking our implementation, it that it's literally _copied_ in their binary, which puts as in a extremely delicate place for compatibility.

Reply:
Furthermore, I agree with @nomisRev in that this exposes too much of the implementation. We shouldn't go into the rabbit hole of adding additional information to functions like this one (why only if it has an error? why not the previous error, or anything else?)